src/providers/WorkflowCore.LockProviders.MySQL/MySqlLockProvider.cs

```diff
@@ -0,0 +1,138 @@
+﻿using System;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using WorkflowCore.Interface;
+using System.Data;
+using System.Collections.Generic;
+using System.Threading;
+using MySql.Data.MySqlClient;
+
+namespace WorkflowCore.LockProviders.MySQL
+{
+    public class MySqlLockProvider : IDistributedLockProvider
```

Comment:
Could we add some tests against a real MySQL instance in the `WorkflowCore.Tests.MySQL` test assembly?
Let's also think about how lease time should be tested.